aws-java-sdk-dynamodb/src/main/java/com/amazonaws/services/dynamodbv2/datamodeling/DynamoDBAtomicIncrementor.java

```diff
@@ -0,0 +1,20 @@
+package com.amazonaws.services.dynamodbv2.datamodeling;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Defines a pattern for annotating fields that are numeric to apply an incremental value to the database via
+ * either calling save, or by calling the mapper method incr which only applies those attributes which intend
+ * to atomically increment for the specified key value.
+ *
+ * Created by matt@mjgreenwood.net on 3/3/17.
+ */
+@DynamoDB
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ElementType.FIELD, ElementType.METHOD})
+public @interface DynamoDBAtomicIncrementor {
+    String attributeName() default "";
```

Comment:
@kiiadi the difference is whether the increment value is dynamic.  In the case of the annotation the value is hard coded - each time thru when increment or save (using APPEND_SET) is called the value +N is hardcoded to be a constant value.  When it's part of the POJO its dynamic and dictated via the client.   The intent of this logic is the write operation is always an increment - that a client cannot 'reset' the value thru a PUT operation.